Python/Product/PythonTools/PythonTools/Intellisense/BufferParser.cs

```diff
@@ -294,48 +259,86 @@ IEnumerable<ITextSnapshot> snapshots
 
                 tasks.Add(Tuple.Create(snapshotGroup.ToArray(), analyzer.SendRequestAsync(
                     new AP.FileUpdateRequest {
-                        fileId = entry.FileId,
+                        documentUri = entry.DocumentUri,
                         updates = updates
                     }
                 )));
             }
 
+            var needRetry = new List<ITextSnapshot>();
             foreach (var task in tasks) {
                 var res = await task.Item2;
 
-                if (res != null) {
-                    Debug.Assert(res.failed != true);
+                if (res?.failed ?? false) {
+                    Interlocked.Decrement(ref analyzer._parsePending);
```

Comment:
Do we need a `finally` block somewhere here to ensure that `_parsePending` is decremented in case of exception?

Reply:
Any exception coming out of the Send call is irrecoverable.

Reply:
Ok